recipes/glfw/all/patches/3.3.2-0003-Wayland-Remove-extra-cmake-modules-dependency.patch

```diff
@@ -0,0 +1,122 @@
+From 8622c81fe5271d10650e4007e96bc5342ac87ec1 Mon Sep 17 00:00:00 2001
+From: Jordan Williams <jordan@jwillikers.com>
+Date: Fri, 3 Nov 2023 08:51:04 -0500
+Subject: [PATCH] Wayland: Remove extra-cmake-modules dependency
+
+Fixes #1774.
+---
+ CMakeLists.txt     | 11 ++++----
+ src/CMakeLists.txt | 67 +++++++++++++++++++++++++++++-----------------
+ 2 files changed, 48 insertions(+), 30 deletions(-)
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index e9888624..33d1a0c1 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -237,12 +237,13 @@ endif()
+ # Use Wayland for window creation
+ #--------------------------------------------------------------------
+ if (_GLFW_WAYLAND)
+-    find_package(ECM REQUIRED NO_MODULE)
+-    list(APPEND CMAKE_MODULE_PATH "${ECM_MODULE_PATH}")
+ 
+-    find_package(Wayland REQUIRED Client Cursor Egl)
+-    find_package(WaylandScanner REQUIRED)
+-    find_package(WaylandProtocols 1.15 REQUIRED)
++    include(FindPkgConfig)
++    pkg_check_modules(Wayland REQUIRED
```

Comment:
What's the problem by using CMakeDeps files?

Reply:
This particular patch was pulled from the upstream project to remove the dependency on the extra-cmake-modules, which doesn't actually work correctly when used with the Conan package. I think it's easier to actually leave this part of the patch intact rather than attempting to minimize the size of the patch, especially since it would require extra work to define things like they are in extra-cmake-modules. In particular, we'd have to define `WAYLAND_PROTOCOLS_PKGDATADIR` ourselves, whereas the pkg-config uses a standard variable already provided by our pkg-config file for the wayland package.